packages/engine/crawlers/sui/index.ts

```diff
@@ -18,33 +17,15 @@ export const suiEngineRunner: SuiRunner = {
 
 		for (const item of key.docs) {
 			const owner = item._id;
-			const { data: allCoins } = await connection.getAllCoins({ owner });
-			let allObjects: SuiObjectResponse[] = [];
+			const allCoinsBalance = await connection.getAllBalances({ owner });
+			const docPromises: Promise<TokenDocument>[] = new Array(
+				allCoinsBalance.length,
+			);
 
-			if (allCoins.length === 0) {
-				allCoins.push({ balance: '0', coinType: suiCoinType } as never);
-				allObjects.push({
-					data: { type: `0x2::coin::Coin<${suiCoinType}>` } as never,
-				});
-			} else {
-				allObjects = await connection.multiGetObjects({
-					ids: allCoins.map((item) => item.coinObjectId),
-					options: { showType: true, showDisplay: true },
-				});
-			}
-
-			const docPromises: Promise<TokenDocument>[] = new Array(allCoins.length);
+			for (let i = 0; i < allCoinsBalance.length; i += 1) {
+				const coin = allCoinsBalance[i];
 
-			for (let i = 0; i < allCoins.length; i += 1) {
-				const coin = allCoins[i];
-				const object = allObjects[i];
-
-				docPromises[i] = getTokenDocument(
-					context,
-					owner,
-					coin,
-					object?.data as never,
-				);
+				docPromises[i] = getTokenDocument(context, owner, coin);
```

Comment:
make sure that we can avoid multiple request, if that is possible then should send as less request as possible